server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -801,6 +808,23 @@ private EngineFactory getEngineFactory(final IndexSettings idxSettings) {
             .filter(maybe -> Objects.requireNonNull(maybe).isPresent())
             .toList();
         if (engineFactories.isEmpty()) {
+            if (indexMetadata == null || isFullySupportedVersion(indexMetadata, MINIMUM_COMPATIBLE)) {
+                return new InternalEngineFactory();
+            } else if (isReadOnlySupportedVersion(indexMetadata, MINIMUM_COMPATIBLE, MINIMUM_READONLY_COMPATIBLE)) {
+                return config -> {
+                    var globalCheckpoint = config.getGlobalCheckpointSupplier().getAsLong();
+                    return new ReadOnlyEngine(
+                        config,
+                        new SeqNoStats(globalCheckpoint, globalCheckpoint, globalCheckpoint),
```

Comment:
Would null not work fine too? I think that does some verification.

Reply:
I think it should work, and +1 for the verification it brings.